Refresh kubeadm docs for Windows (#33324)
kubernetes/website: content/en/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm.md

  kubeadm join --token <token> <control-plane-host>:<control-plane-port> --discovery-token-ca-cert-hash sha256:<hash>
  ```

{{< note >}}

> On Windows you may need to specify the `cri-socket` parameter.  For example, with Containerd as the runtime add `--cri-socket "npipe:////./pipe/containerd-containerd"` to the `kubeadm join` command

Reviewer: When do users not need to specify the `cri-socket` parameter for Windows?
If we say `may need to` I think we should give guidance on when they would or wouldn't need to specify the extra arg.

Author: Makes sense.  I believe it is required for any remote runtimes.  If using docker shim then it is not required.  Since these docs are for 1.24+ should I be more direct and be more direct that it is requird something like:

```suggestion
On Windows when using a remote runtime the `cri-socket` parameter is required.  For example when using Containerd as the runtime add `--cri-socket "npipe:////./pipe/containerd-containerd"` to the `kubeadm join` command
```

Reviewer 2: I think from Tuesday (the expected v1.24 release date), setting a CRI socket path becomes mandatory.

Author: Is this required then?

Reviewer: Maybe we should say:

On Windows you should specify the `cri-socket` parameter in the `kubeadm join` command. The default path is "npipe:////./pipe/containerd-containerd" for containerd.

Author: updated

  kubeadm join --token <token> <control-plane-host>:<control-plane-port> --discovery-token-ca-cert-hash sha256:<hash>
  ```

{{< note >}}

> On Windows you should specify the `--cri-socket` parameter in the `kubeadm join` command. The default path is "npipe:////./pipe/containerd-containerd" for containerd.

Reviewer 2: _default path_? This doesn't seem quite right.

Are we saying “if you chose containerd, then containerd is likely to make its socket available as `npipe:////./pipe/containerd-containerd`?

If so, that's not a default. It's not even a recommendation (we don't recommend a container runtime).

Reviewer: That path is the value used by containerd if there is no gprc address specified to containerd (either in the config.toml or with `--address`/`-a` .

Reviewer 2: OK. We should not document this as if it is a Kubernetes default.

Reviewer: I think we should keep the note that on Windows (at least for now) the path needs to be specified but agree we can just say something like 'On Windows you should specify the `--cri-socket` parameter in the `kubeadm join` command to the gRPC path exposed by your container runtime.'

Reviewer: Actually I don't think we need to keep the note.
https://github.com/kubernetes/kubernetes/pull/107317 updated the search paths used by kubelet to auto-detect the correct CRI socket ~5 months ago.
This should be sufficient.

Author: I've removed this note since it will be automatically discovered
